src/main/java/it/gov/pagopa/pu/workflow/wf/exportfile/wfexportfile/ExportFileWFImpl.java

```diff
@@ -0,0 +1,115 @@
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+@Slf4j
+@WorkflowImpl(taskQueues = ExportFileWFImpl.TASK_QUEUE_EXPORT_FILE_WF)
+public class ExportFileWFImpl implements ExportFileWF, ApplicationContextAware {
+  public static final String TASK_QUEUE_EXPORT_FILE_WF = "ExportFileWF";
+  public static final String TASK_QUEUE_EXPORT_FILE_LOCAL_ACTIVITY = "ExportFileWF_LOCAL";
+
+  @Value("${schedule.export-file-expiration.days}")
+  private int expirationDays;
+
+  private ExportFileActivity exportFileActivity;
+  private UpdateExportFileStatusActivity updateExportFileStatusActivity;
+  private ScheduleExportFileExpirationActivity scheduleExportFileExpirationActivity;
+
+  /**
+   * Temporal workflow will not allow to use injection in order to avoid <a href="https://docs.temporal.io/workflows#non-deterministic-change">non-deterministic changes</a> due to dynamic reconfiguration.<BR />
+   * Anyway it allows to override ActivityOptions, but actually it's not supporting the override based on the particular workflow.<BR />
+   * In {@link it.gov.pagopa.pu.workflow.config.TemporalWFImplementationCustomizer} we are already setting defaults to all workflows.<BR />
+   * Use this as an example to override based on the particular workflow.
+   */
+  @Override
+  public void setApplicationContext(ApplicationContext applicationContext) throws BeansException {
+    ExportFileWFConfig wfConfig = applicationContext.getBean(ExportFileWFConfig.class);
+    exportFileActivity = wfConfig.buildExportFileActivityStub();
+    updateExportFileStatusActivity = wfConfig.buildUpdateExportFileStatusActivityStub();
+    scheduleExportFileExpirationActivity = wfConfig.buildScheduleExportFileExpirationActivityStub();
+  }
+
+  @Override
+  public void exportFile(Long exportFileId, ExportFileTypeEnum exportFileType) {
+    log.info("Starting export file workflow for exportFileId: {}", exportFileId);
+    boolean success = processExportFile(exportFileId, exportFileType);
+    if(success){
+      scheduleExportFileExpiration(exportFileId);
+    }
+    //TODO sendEmailActivity will be added with the task P4ADEV-2597
+    log.info("Completed export file workflow for exportFileId: {}", exportFileId);
+  }
+
+  private boolean processExportFile(Long exportFileId, ExportFileTypeEnum exportFileType) {
+    updateExportFileStatus(UpdateStatusRequest.builder()
+      .exportFileId(exportFileId)
+      .oldStatus(ExportFileStatus.REQUESTED)
+      .newStatus(ExportFileStatus.PROCESSING)
+      .build());
+    Pair<ExportFileResult,String> exportResultPair = executeExport(exportFileId, exportFileType);
+    ExportFileResult exportFileResult = exportResultPair.getLeft();
+    String errorDescription = exportResultPair.getRight();
+    updateExportFileWithProcessingResult(exportFileId, errorDescription, exportFileResult);
```

Comment:
move these instructions on the executeExport method in order to clearly see the activity invocations which compose the WF